Neo4j.Driver/Neo4j.Driver/Internal/AsyncSessionResourceManager.cs

```diff
@@ -71,15 +71,15 @@ public Task OnTransactionDisposeAsync(Bookmarks bookmarks, string database)
         /// <param name="bookmarks">The new bookmarks.</param>
         public void UpdateBookmarks(Bookmarks bookmarks, IDatabaseInfo dbInfo = null)
         {
+            _initialBookmarks = null;
             if (bookmarks == null || !bookmarks.Values.Any())
                 return;
 
             var previousBookmarks = _bookmarks?.Values ?? Array.Empty<string>();
             _bookmarks = bookmarks;
 
-            var db = dbInfo?.Name ?? _database;
             if (_useBookmarkManager)
-                _bookmarkManager.UpdateBookmarksAsync(db, previousBookmarks, bookmarks.Values);
+                _bookmarkManager.UpdateBookmarksAsync(previousBookmarks, bookmarks.Values).GetAwaiter().GetResult();
```

Comment:
this can now block until bookmark manager has completed update, including user callback.